app/hooks/useChatWithVision.ts

```diff
@@ -0,0 +1,24 @@
+import OpenAI from "openai";
+import { useMutation, UseMutationResult } from "@tanstack/react-query";
+import axios from "axios";
+import type { ChatWithVisionVariables } from "@/lib/types";
+
+type Completions = OpenAI.Beta.Chat.Completions;
+
+const mutationFn = async (
+  args: ChatWithVisionVariables
+): Promise<Completions> => {
+  const { data } = await axios.post<Completions>("/api/chat-with-vision", args);
+  return data;
+};
+
+export const useChatWithVision = (): UseMutationResult<
```

Comment:
in the consuming component I use like this
```
  const chatWithVision = useChatWithVision() ;
  .
  .
  .
            {chatWithVision?.isPending ? <CircularProgress /> : <SendIcon />}

  

```

and I don't have to worry about storing loading state somewhere and resetting it